package.json

```diff
@@ -86,7 +86,7 @@
     "tslint-config-prettier": "^1.18.0",
     "typedoc": "0.15.8",
     "typedoc-plugin-markdown": "2.2.11",
-    "typescript": "^3.8.3",
```

Comment:
I think [TypeScript 5.4 will give us a lot of useful tools](https://devblogs.microsoft.com/typescript/announcing-typescript-5-4-beta/#the-noinfer-utility-type) like `NoInfer`.

Also changing this to be a literal version, because [TypeScript doesn't follow semantic versioning](https://www.learningtypescript.com/articles/why-typescript-doesnt-follow-strict-semantic-versioning), so we can't rely on things like major version locking to keep TS behavior stable.